### PR TITLE
Add limited support for aarch64 and ppc64le using qemu on azure

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This is based on #882

### CODE REVIEW

File: conda_smithy/templates/azure-pipelines-osx.yml.tmpl [hunk 1]
```diff
@@ -6,10 +6,11 @@ jobs:
   strategy:
     maxParallel: 8
     matrix:
-    {%- for config, platform in configs | sort %}
+    {%- for config, platform, upload in configs | sort %}
     {%- if platform.startswith('osx') %}
       {{ config }}:
-        CONFIG: {{ config}}
+        CONFIG: {{ config }}
+        UPLOAD_PACKAGES: {{ upload }}
```
Comment (isuruf) [inline conda_smithy/templates/azure-pipelines-osx.yml.tmpl]: @mariusvniekerk, is there a way to make this a global env variable?

Comment (mariusvniekerk) [reply conda_smithy/templates/azure-pipelines-osx.yml.tmpl]: that looks pretty global to me?

File: conda_smithy/configure_feedstock.py [hunk 2]
```diff
@@ -915,13 +900,13 @@ def _load_forge_config(forge_dir, exclusive_config_file):
                   'linux': 'circle', 
                   'osx': 'travis', 
                   'win': 'appveyor',
-                  'linux_aarch64': 'azure',
-                  'linux_ppc64le': 'azure',
+                  # Following platforms are disabled by default
+                  'linux_aarch64': None,
```
Comment (mariusvniekerk) [inline conda_smithy/configure_feedstock.py]: technically the enabled switch is on line 910

File: conda_smithy/templates/build_steps.sh.tmpl [hunk 3]
```diff
@@ -36,8 +36,8 @@ make_build_number "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
 conda build "${RECIPE_ROOT}" -m "${CI_SUPPORT}/${CONFIG}.yaml" \
     --clobber-file "${CI_SUPPORT}/clobber_${CONFIG}.yaml"
 
-{% if upload_packages %}
-upload_package "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
-{% endif %}
+if [[ "${UPLOAD_PACKAGES}" != "False" ]]; then
```
Comment (mariusvniekerk) [inline conda_smithy/templates/build_steps.sh.tmpl]: we need to ensure that this env var is set by the run docker steps phase

Comment (mariusvniekerk) [reply conda_smithy/templates/build_steps.sh.tmpl]: DONE

File: conda_smithy/templates/azure-pipelines-osx.yml.tmpl (hunk 1 above)
Comment (mariusvniekerk) [reply conda_smithy/templates/azure-pipelines-osx.yml.tmpl]: so we could use a `variables` block

File: conda_smithy/configure_feedstock.py (hunk 2 above)
Comment (isuruf) [reply conda_smithy/configure_feedstock.py]: Those enabled switches are not supposed to be overridden by the users. They are used for the Readme etc. Overriding using
```
provider:
  ppc64le: azure
```
should work

**Pull request merged**